test/controllers/account/manage_permissions_controller_test.rb

```diff
@@ -30,6 +30,20 @@ class Account::ManagePermissionsControllerTest < ActionController::TestCase
           assert_select "td", count: 1, text: non_delegatable_no_access_to_app.name
         end
       end
+
+      should "not list retired applications" do
+        user = create(:admin_user, email: "admin@gov.uk")
+        sign_in user
+
+        retired_app = create(:application, retired: true)
+        user.grant_application_signin_permission(retired_app)
+
+        get :show
+
+        assert_select ".container" do
+          assert_select "td", count: 0, text: retired_app.name
```

Comment:
Is it possible to drop the `text` parameter from this assertion? It might make the test even stronger as I don't think we should expect any `td` given the setup?

Reply:
While you're probably right that the test setup means there won't be any rows in the table, I think the only thing I care about in this test is that the retired app doesn't show up. And I think asserting that explicitly makes the intention of the test clearer. So I think I'm going to leave this as it is.